src/index.tsx

```diff
@@ -42,21 +42,39 @@ const Typer: React.FC<TyperProps> = ({
   const [isDeleting, setIsDeleting] = useState(false);
   const [loopNum, setLoopNum] = useState(0);
   const [writingSpeed, setWritingSpeed] = useState(typingSpeed);
+  const mountedRef = useRef(false);
 
   const refText = useRef(text);
   const refIsDeleting = useRef(isDeleting);
   const refLoopNum = useRef(loopNum);
   const refWritingSpeed = useRef(writingSpeed);
 
   const refTimer = useRef(0);
+  const timer1 = useRef(0);
+  const timer2 = useRef(0);
+  const timer3 = useRef(0);
   const refIsGoingToDelete = useRef(false);
   const refIsFinished = useRef(false);
 
+  clearTimeout(refTimer.current);
```

Comment:
I found you need to cleat out this main timer on every re-render. Otherwise, you might get into inconsistent speed for the typer specially on other components re-rendring.